### DISCUSSION TITLE
PolyGlot: Language Construction Toolkit 3.5

### DISCUSSION THREAD

**Discussion by DraqueT** [DraqueT/PolyGlot#1127] [General]
Heyo, all! Welcome to version 3.5 of PolyGlot! This is a **massive** release! First, I want to give a huge shout out to TrapinchO over on GitHub, who gave an enormous amount of help with testing, and just has killer ideas in general! 3.5 includes a long list of upgrades and bug fixes. This also represents a significant step toward an Android release of PolyGlot, which has been much requested and a long time coming.
                            
Among the most exciting upgrades are the complete integration of the Zompist word generator (algorithm and original design by Mark Rosenfelder there), a complete overhaul of how graphics are painted (no more CPU fans going nuts), a revamped lexicon look (local language synonyms now displayed in the list by their conword counterparts), automatic syllable composition when generating pronunciations, and many, _many_ quality of life improvements (full list below). And that is on top of a ton of bug fixes!

Check Language upgrade 
 - Check Languge now automatically checks to see whether any characters unsupported by your current fonts are used in your language. Should be helpful to anyone using a custom script.

PDF printing now accepts/uses local language font
 - Previously PDF printing did not read local lang fonts at all

If present, romanized forms of words will export to Excel
 - Previously these values were ignored

Tooltips now automatically format in a way that is much nicer to look at
 - Auto linebreaks added for better readability.

Font compatibility in PDF printing significantly improved
 - Added in a library that can convert fonts to more readable formats when necessary.

Reworked printing of word class values to PDF
 - Word classes now print more cleanly to PDF.
 
PolyGlot now handles the awfulness that is the Windows Fonts folder correctly
 - It's this insane virtual folder unlike anything else I've seen in the Windows system. 
 
Startup time reduced
 - Added quite a few optimizations to make PolyGlot boot faster.
 
Upgraded combobox displays
 - Now display the field label even when a value is selected, and if the value is a word, its localword equivalent is displayed next to choices
 
Dropdowns now filter as you begin to type
 - If you select a combobox and begin typing, the displayed choices will filter based on matches
 
BIG update to core functionality to allow for development of Android app
 - Y'all seem to want this like crazy. Getting there.

Upgraded to Java 17 - Long Term Home for PolyGlot (no more Java upgrades until next LTS)
 - Won't matter much to most users.

New easter egg added.
 - owo

BUGS FIXED
- Ligatures loaded initially, but failed to re-load from saved PolyGlot archive
- Broken multi-delete in conjugations menu fixed
- Graphical artifacting and "shadows" appeared sometimes in etymology window
- Excel import bugs corrected (false success report)
- Quickentry image insertion caused PolyGlot to freeze
- Quiz could make copies of the correct answer (with copies being "wrong")
- Local languge sizing failed to function properly in menus
- IPA Conversion tool converted text with HTML interspersed
- If no alphabet is defined at all, "check language" feature crashes program
- Under certain circumstances, text boxes could be mistakenly set to the conlang font
- Search menu populates font and size options from wrong place
- Hitting the filter button while is already applied did nothing
- Deletion of top level etymological parent caused unhandled exception
- Excel export applied conjugation transforms without regard to rules
- Excel export did not properly set conlang font on conjugated wordform cells
- Excel export sometimes printed empty tabs
- Deleting an internal etymological parent resized elements of the etymology window
- If you had too many word classes, it would break the autodeclension setup menu
- Deleting an entry in the phonemic orthography menu would also delete any entries with the same values
- Elements of the conjugation menu were failing to render in the appropriate font
- The grammar chapter section could become persistently wonky if multiple chapters without names were added in a row
- Fixed menus that could display user text but which did not use local language font (possible tofu characters)
- Part of speech dropdown on Lexicon did not respect font updates
- Old JSoup version had serious security bug. Upgraded to plug.
- Fixed various lexicon filter bugs

<hr /><em>This discussion was created from the release <a href='https://github.com/DraqueT/PolyGlot/releases/tag/3.5'>PolyGlot: Language Construction Toolkit 3.5</a>.</em>

**Comment by sTiKyt** [comment]
Just updated AUR package at:
https://aur.archlinux.org/packages/draquet-polyglot-bin/

**Reply by DraqueT** [reply]
Thank you! :D

**Reply by sTiKyt** [reply]
> Thank you! :D

No, thank you ;D
 this tool is only of it's kind as far as i was able to find, very good stuff
